### PR TITLE
add fuctions( getIntrinsics and getImageResolution ) to ARKitController

### PR DESCRIPTION
**Pull request by Oct7**
Hello

I've added functions that get camera intrinsics or image resolution for getting some raw datas. I believe this addition could be beneficial for the community. Please find my pull request attached for your review.

Best regards,
Oct7

### CODE REVIEW

File: ios/Classes/FlutterArkitView+Handlers.swift [hunk 1]
```diff
@@ -245,6 +245,24 @@ extension FlutterArkitView {
       result(nil)
     }
   }
+    func onCameraIntrinsics(_ result: FlutterResult) {
```
Comment (olexale) [inline ios/Classes/FlutterArkitView+Handlers.swift:248]: Please reformat this file with [swiftformat](https://github.com/nicklockwood/SwiftFormat).

File: lib/src/utils/json_converters.dart [hunk 2]
```diff
@@ -27,18 +27,15 @@ class StringValueNotifierConverter extends ValueNotifierConverter<String> {
   const StringValueNotifierConverter() : super();
 }
 
-class ListMaterialsValueNotifierConverter
-    implements JsonConverter<ValueNotifier<List<ARKitMaterial>?>, List<Map>?> {
+class ListMaterialsValueNotifierConverter implements JsonConverter<ValueNotifier<List<ARKitMaterial>?>, List<Map>?> {
```
Comment (olexale) [inline lib/src/utils/json_converters.dart:30]: Please reformat this and other Dart files with the default "80 characters" line width.

**Review by olexale — COMMENTED** [review]
First of all, thank you so much for the PR! I don't have much time to work on this project, so I'm happy to see that the community members like you drive it forward!

The PR looks good, but please reformat the code with the default Dart formatting.

As this project lacks integration tests, would please create a page for the example project that demonstrates how to use this new functionality? I use these samples for smoke testing before pushing any new version to make sure that in the new version everything works as expected.

**Review by Oct7 — COMMENTED** [review]
Hi there! 🥳

I'm happy to report that I've updated the code based on your feedback. Soon, I'll add new options to the captureHighResolutionFrame function in ARKit. For more details, check out the [ARKit Documentation](https://developer.apple.com/documentation/arkit/arsession/3975720-capturehighresolutionframe).

I've noticed that the imageResolution option previously added doesn't quite match the default capturedImage resolution from the controller.snapshot. I'm addressing this for better compatibility. Also, I'll update new examples for getImageResolution, getIntrinsics, and captureHighResolutionFrame when I update captureHighResolutionFrame function.

Thanks for your support. Looking forward to sharing these updates!

**Comment by olexale** [comment]
Hey! 👋
Thank you for your updates! Would you please address the last comment from my initial review and provide a sample for the added functionality? It would help others to understand how to use it + will allow us to double-check that nothing is broken when we release a new version. 
Thanks in advance!

**Comment by Oct7** [comment]
I've been so busy with my main job that I've postponed this matter for a long time. I apologize for this. I've added the example files in the way you originally used. Please check them.
thank you :)

**Comment by olexale** [comment]
Hello @Oct7,
Thanks again for your contribution! I've made a few changes to your PR. You may expect a new version on pub.dev soon.

**Pull request merged**